src/shared/services/bookmarks-views-plays-service.ts

```diff
@@ -0,0 +1,329 @@
+				contentUuid,
+				user,
+			});
+			if (silent) {
+				console.error(error);
+			} else {
+				throw error;
+			}
+		}
+	}
+
+	public static async getItemCounts(
+		itemUuid: string,
+		user: Avo.User.User | null
+	): Promise<BookmarkViewPlayCounts> {
+		const response = await dataService.query({
+			query: GET_ITEM_BOOKMARK_VIEW_PLAY_COUNTS,
+			variables: { itemUuid, profileId: get(user, 'profile.id', null) },
+		});
+		const isBookmarked = !!get(response, 'data.app_item_bookmarks[0]');
+		const bookmarkCount = get(response, 'data.app_item_bookmarks_aggregate.aggregate.count', 0);
+		const viewCount = get(response, 'data.app_item_views[0].count', 0);
+		const playCount = get(response, 'data.app_item_plays[0].count', 0);
+		return {
+			bookmarkCount,
+			viewCount,
+			playCount,
+			isBookmarked,
+		};
+	}
+
+	public static async getCollectionCounts(
+		collectionUuid: string,
+		user: Avo.User.User | null
+	): Promise<BookmarkViewPlayCounts> {
+		const response = await dataService.query({
+			query: GET_COLLECTION_BOOKMARK_VIEW_PLAY_COUNTS,
+			variables: { collectionUuid, profileId: get(user, 'profile.id', null) },
+		});
+		const isBookmarked = !!get(response, 'data.app_collection_bookmarks[0]');
+		const bookmarkCount = get(
+			response,
+			'data.app_collection_bookmarks_aggregate.aggregate.count',
+			0
+		);
+		const viewCount = get(response, 'data.app_collection_views[0].count', 0);
+		const playCount = get(response, 'data.app_collection_plays[0].count', 0);
+		return {
+			bookmarkCount,
+			viewCount,
+			playCount,
+			isBookmarked,
+		};
+	}
+
+	/**
+	 * Toggles the bookmark for the provided item or collection or bundle
+	 * @param contentId
+	 * @param user
+	 * @param type
+	 * @param isBookmarked
+	 * @return {boolean} returns true of the operation was successful, otherwise false
+	 */
+	public static async toggleBookmark(
+		contentId: string,
+		user: Avo.User.User,
+		type: EventContentType,
+		isBookmarked: boolean
+	): Promise<boolean> {
+		try {
+			if (!contentId) {
+				throw new CustomError(
+					`Failed to bookmark ${type} because the ${type} doesn't seem to be loaded yet`,
+					null,
+					{ contentId }
+				);
+			}
+			await BookmarksViewsPlaysService.action(
+				isBookmarked ? 'unbookmark' : 'bookmark',
+				type,
+				contentId,
+				user,
+				false
+			);
+			return true;
+		} catch (err) {
+			console.error('Failed to bookmark/unbookmark the item', err, { contentId });
+			toastService.danger(
+				isBookmarked
+					? i18n.t('Het aanmaken van de bladwijzer is mislukt')
+					: i18n.t('Het verwijderen van de bladwijzer is mislukt')
+			);
+			return false;
+		}
+	}
+
+	/**
+	 * Gets all bookmarks for user without pagination
+	 * since we cannot order items across both tables: item_bookmarks and collection_bookmarks
+	 * @param user
+	 */
+	public static async getAllBookmarksForUser(user: Avo.User.User): Promise<BookmarkInfo[]> {
+		const response: GetBookmarksForUserResponse = await dataService.query({
```

Comment:
Idem?

Reply:
die exceptie wordt opgevangen in Bookmarks.tsx